project_cost_report/static/src/js/report_action.js

```diff
@@ -0,0 +1,386 @@
+        }
+        return this.controlPanelButtons;
+    },
+    /**
+     * Update the control pannel of the report view.
+     */
+    updateControlPanel(){
+        this.update_control_panel({
+            breadcrumbs: this.getParent().get_breadcrumbs(),
+            cp_content: {$buttons: this.getControlPanelButtons()},
+        });
+    },
+    /**
+     * Hide/show the FOLD/UNFOLD buttons given the folding state.
+     *
+     * The folding state is managed with the attribute `unfolded`.
+     * This method must be called after setting the attribute
+     * in order to update the DOM nodes.
+     */
+    updateFoldButtonVisibility(){
+        if(this.unfolded){
+            this.foldButton.show();
+            this.foldButton.css('display', 'inline-block');
+            this.unfoldButton.hide();
+        }
+        else {
+            this.foldButton.hide();
+            this.unfoldButton.show();
+            this.unfoldButton.css('display', 'inline-block');
+        }
+    },
+    /**
+     * Start the widget.
+     *
+     * This is a standard method of web.Widget.
+     * It is called when the widget is ready to be rendered.
+     */
+    start(){
+        var result = this._super();
+        this.updateControlPanel();
+        this.updateHtml();
+        this.updateFoldButtonVisibility();
+        return result;
+    },
+    /**
+     * Refresh the report.
+     */
+    async refresh(){
+        this.updateHtml();
+        this.updateFoldButtonVisibility();
+    },
+    /**
+     * Download the PDF version of the report.
+     */
+    downloadPDF(){
+        framework.blockUI();
+        session.get_file({
+            url: "/web/project_cost_report/" + this.projectId,
+            data: {report_context: JSON.stringify(this.reportContext)},
+            complete: framework.unblockUI,
+            error: crash_manager.rpc_error.bind(crash_manager),
+        });
+    },
+    /**
+     * Fold all report categories.
+     */
+    fold(){
+        this.reportContext.unfolded_categories = {
+            'product': [],
+            'time': [],
+            'outsourcing': [],
+        };
+        this.unfolded = false;
+        this.refresh();
+    },
+    /**
+     * Unfold all report categories.
+     */
+    async unfold(){
+        this.reportContext.unfolded_categories = await this._rpc({
+            model: 'project.cost.report',
+            method: 'get_foldable_categories',
+            args: [this.projectId],
+        });
+        this.unfolded = true;
+        this.refresh();
+    },
+    /**
+     * Fold a single category.
+     *
+     * @param {Number} categoryId - the ID of the category.
+     * @param {String} sectionName - the name of the section.
+     */
+    foldCategory(categoryId, sectionName){
+        this.reportContext.unfolded_categories[sectionName] = (
+            this.reportContext.unfolded_categories[sectionName].filter((c) => c != categoryId)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected '!==' and instead saw '!='. (eqeqeq)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3675949)